_posts/2024-02-23-enhanced-multi-vector-support-in-opensearch-knn.md

```diff
@@ -0,0 +1,182 @@
+
+<img src="/assets/media/blog-images/2024-02-23-multi-vector-support-in-knn/multi-vector-before-2.png" alt="multi-vector-behavior-before-cluster-level" class="img-centered"/>
+
+## Improvements in OpenSearch 2.12
+In the latest version, OpenSearch leverages document and vector mapping data to deduplicate search results. This process occurs when vectors belonging to the same document are already collected in the search queue. During deduplication, OpenSearch selects the vector closest to the query vector and discards the farther one. This value is then converted back to an OpenSearch search score, ultimately becoming the document's score.
+
+Let’s take a look at an example with the diagrams below. Consider two documents, Document 1 and Document 2. Document 1 contains vectors A and B, while Document 2 contains vectors C and D (Fig. 1). During the search, vector A is found and added to the search queue (Fig. 2). Subsequently, vector B is encountered. Since both vectors A and B belong to Document 1, their distances are compared. The distance between the query vector and vector B is 0.9, which is less than the distance between the query vector and vector A, which is 1.0. As a result, vector A is removed from the search queue, and vector B is added (Fig. 3).
+
+Moving forward, vector D is discovered. Since vector D belongs to Document 2, whose vector is not in the search queue, it is added (Fig. 4). Then, vector C is found. As vector C belongs to Document 2, whose vector is already in the search queue, the distances between the query vector and vectors C and D are compared. The distance between the query vector and vector D (1.1) is less than that between the query vector and vector C (1.2), so vector D remains in the search queue (Fig. 5).
+
+In the end, both Document 1 and Document 2 are returned, and their scores are calculated based on the distances between the query vector and the vectors collected in the search queue.
+
+<img src="/assets/media/blog-images/2024-02-23-multi-vector-support-in-knn/multi-vector-after-1.png" alt="multi-vector-behavior-after" class="img-centered"/>
+
+Now, even with a k value set to 2, the search will return 2 documents, regardless of whether the 2 nearest vectors belong to one or multiple documents.  Additionally, this enhancement improves recall for users, as each segment now returns the k nearest documents instead of just the k nearest vectors of documents.
+## How to use nested field to store multi-vector
+To utilize nested fields for storing multi-vectors, ensure that your OpenSearch cluster has the KNN plugins installed. Now, let's delve into the process of creating a KNN index with nested fields and conducting searches on it.
```

Comment:
I don't think we should call out installing the k-NN plugin because it is installed by default and presumably the users reading this blog will be familiar with k-NN already.
```suggestion
Let's delve into the process of creating a k-NN index with nested fields and conducting searches on it.
```